equinox/nn/_kv_cache.py

```diff
@@ -0,0 +1,50 @@
+import jax.lax as lax
+import jax.numpy as jnp
+from jaxtyping import Array, Float, Int
+
+from .._misc import default_int_dtype
+from .._module import field, Module
+from ._stateful import State, StateIndex
+
+
+class StandardKVCache(Module):
+    key_shape: tuple[int, int, int] = field(static=True)
+    value_shape: tuple[int, int, int] = field(static=True)
+
+    autoregressive_index: StateIndex
+
+    def __init__(
+        self,
+        key_shape: tuple[int, int, int],
+        value_shape: tuple[int, int, int],
+    ):
+        self.key_shape = key_shape
+        self.value_shape = value_shape
+
+        def _make_cache(**_):
+            _int = default_int_dtype()
+            return jnp.empty(key_shape), jnp.empty(value_shape), jnp.zeros((), _int)
+
+        self.autoregressive_index = StateIndex(_make_cache())
```

Comment:
This function is defined and then immediately called. (I recognise this as the old pattern we had for `StateIndex`!) Can we inline it instead?